### PR TITLE
feat: Review page

### LINKED ISSUE #592: Review screen
I'm responsible for reviewing many goals and I'm having a hard time catching up at the turn of the month.

My tools available are
- email inbox
- global goal hierarchy
- notifications

all three are very cluttered (notifications I now completely ignore tbh).

What I need is a clean list of things I **must do**, like status updates for goals I own + what I need to acknowledge.

Previously we had some discussions about how this plays with tasks but I think we need a simple zero to one solution first.

### PR DESCRIPTION
**Pull request by Rockyy174**
I've added the new Review page where champions and reviewers can easily see all their due assignments.

I'm marking this PR as a draft since it must be merged after https://github.com/operately/operately/pull/768 and  https://github.com/operately/operately/pull/771.

### CODE REVIEW

File: assets/js/layouts/DefaultLayout/Review.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        fontWeight: "900",
+      }}
+      data-test-id="review-link-count"
+    >
+      {count}
+    </div>
+  );
+}
+
+function useCount() {
```
Comment (shiroyasha) [inline assets/js/layouts/DefaultLayout/Review.tsx:51]: Let's create a model for assignments in `assets/js/models/assignments` and move this logic there. Models are usually a good mediator between the view part and the API, where we can store other data fetching and transformation logic.

As a rule of thumb, I would suggest avoiding the direct use of the api namespace in views directly, and almost always using the models as a middleman.

File: assets/js/pages/ReviewPage/page.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+import { IconTarget, IconHexagons } from "@tabler/icons-react";
+
+import { Paths } from "@/routes/paths";
+import { ReviewAssignment } from "@/api";
+import { useLoadedData } from "./loader";
+import { useNavigateTo } from "@/routes/useNavigateTo";
+
+
+type AssignmentType = "goal" | "project" | "goal_update" | "check_in";
```
Comment (shiroyasha) [inline assets/js/pages/ReviewPage/page.tsx:13]: This is also a good candidate to be moved to the model.

File: assets/js/pages/ReviewPage/page.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+
+function formatDate(dateString: string) {
+  const date = new Date(dateString);
+  const options = { month: 'long', day: 'numeric' } as Intl.DateTimeFormatOptions;
+  return date.toLocaleDateString('en-US', options);
+}
+
+function calculateDaysAgo(dateString: string) {
```
Comment (shiroyasha) [inline assets/js/pages/ReviewPage/page.tsx:123]: Move this to utils/time to centralize time related operations.

File: assets/js/pages/ReviewPage/page.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      )}
+    </div>
+  );
+}
+
+
+function formatDate(dateString: string) {
+  const date = new Date(dateString);
+  const options = { month: 'long', day: 'numeric' } as Intl.DateTimeFormatOptions;
+  return date.toLocaleDateString('en-US', options);
```
Comment (shiroyasha) [inline assets/js/pages/ReviewPage/page.tsx:120]: There is a FormattedTime component, can we use it here?

File: assets/js/layouts/DefaultLayout/Review.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  const [count, setCount] = useState(0);
```
Comment (shiroyasha) [inline assets/js/layouts/DefaultLayout/Review.tsx:52]: The `make gen` is also generating a hook for the query, called useGetAssignementCount that returns a refetch command as well. Can we use it here?

Pseudocode:

```
const { data, refetch } = useGetAssignmentCount();
const count = data.count;

useAssignmentsCount(() => refetch());
```

File: assets/js/layouts/DefaultLayout/Review.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useEffect, useState } from "react";
+
+import { getAssignmentsCount } from "@/api";
+import { useAssignmentsCount } from "@/api/socket";
```
Comment (shiroyasha) [inline assets/js/layouts/DefaultLayout/Review.tsx:4]: Note for the some future development:

Events from the backend should have a name that resembles a signal or a notification, to make the code a bit more clear. Something like `useOnAssignmentCountChange` or `useAssignmentCountChangeSignal`. Afaik, the back-and implementation of sockets is not yet allowing this so we can't do it in this PR, but its a good food for thought in upcoming ones.

File: assets/js/layouts/DefaultLayout/Review.tsx (hunk 5 above)
Comment (Rockyy174) [reply assets/js/layouts/DefaultLayout/Review.tsx:52]: Yes, that was right. I was able to simply the hook quite a bit by using `useGetAssignmentCount`. Thanks!

File: assets/js/pages/ReviewPage/page.tsx (hunk 4 above)
Comment (Rockyy174) [reply assets/js/pages/ReviewPage/page.tsx]: Yes, it was possible to use it. I didn't know about this component. Thank you!

**Comment by Rockyy174** [comment]
I believe everything has been taken care of. Thank you for all the insights, @shiroyasha!

If there is anything else that can be improved, just let me know!

**Pull request merged**